statistics/histogram.go

```diff
@@ -2280,3 +2274,44 @@ func MergePartitionHist2GlobalHist(sc *stmtctx.StatementContext, hists []*Histog
 	}
 	return globalHist, nil
 }
+
+// ColLoadedStatus indicates the status of column statistics
+type ColLoadedStatus struct {
+	wasLoaded         bool
+	cmsEvicting       bool
+	topnEvicting      bool
+	histogramEvicting bool
+}
+
+// NewColFullLoadStatus returns the status that the column fully loaded
+func NewColFullLoadStatus() ColLoadedStatus {
+	return ColLoadedStatus{
+		wasLoaded: true,
+	}
+}
+
+// WasLoaded indicates whether the column's statistics was loaded from storage before.
+// Note that `wasLoaded` only can be set in initializing
+func (s ColLoadedStatus) WasLoaded() bool {
+	return s.wasLoaded
+}
+
+// IsNeedLoaded indicates whether it needs load statistics during LoadNeededHistograms or sync stats
+// If the column was loaded and any statistics of it is evicting, it also needs re-load statistics.
+func (s ColLoadedStatus) IsNeedLoaded() bool {
```

Comment:
What about naming it as "isLoadNeeded" or "needToLoad"?

Reply:
updated.